dev/src/index.ts

```diff
@@ -808,10 +807,6 @@ export class Firestore implements firestore.Firestore {
     }
 
     if (createEnabledInstance) {
-      // Re-use the existing EnabledTraceUtil if one has been created.
-      if (this._traceUtil && this._traceUtil instanceof EnabledTraceUtil) {
-        return this._traceUtil;
-      }
```

Comment:
Since we are now capturing settings as part of the trace attributes, if the settings change we need to also change the traceUtil instance. So we should _not_ re-use the existing one.